client/state/jetpack-settings/modules/reducer.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import assign from 'lodash/assign';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SERIALIZE,
+	DESERIALIZE,
+	JETPACK_MODULE_ACTIVATE,
+	JETPACK_MODULE_ACTIVATE_FAIL,
+	JETPACK_MODULE_ACTIVATE_SUCCESS
+} from 'state/action-types';
+
+/**
+ * `Reducer` function which handles request/response actions
+ * concerning Jetpack modules data updates
+ *
+ * @param  {Array}  state  Current state
+ * @param  {Object} action action
+ * @return {Array}         Updated state
+ */
+export const items = ( state = {}, action ) => {
+	switch ( action.type ) {
+		case JETPACK_MODULE_ACTIVATE_SUCCESS:
+			return assign( {}, state, {
+				[ action.module ]: assign( {}, state[ action.module ], { activated: true } )
+			} );
+		// return initial state when serializing/deserializing
+		case SERIALIZE:
+		case DESERIALIZE:
+			return {};
+		default:
+			return state;
+	}
+};
+
+export const initialRequestsState = {
+	activating: {},
+};
+
+/**
+ * `Reducer` function which handles request/response actions
+ * concerning Jetpack modules-related requests
+ *
+ * @param {Object} state - current state
+ * @param {Object} action - action
+ * @return {Object} updated state
+ */
+export const requests = ( state = initialRequestsState, action ) => {
```

Comment:
Here we'll also have to track modules by `siteId`. Using the above suggestions, I'd change it to:

```
export const requests = ( state = initialRequestsState, { type, siteId, module } ) => {
	switch ( type ) {
		case JETPACK_MODULE_ACTIVATE:
			return merge( {}, state, {
				[ siteId ]: {
					[ module ]: true
				}
			} );
		case JETPACK_MODULE_ACTIVATE_FAIL:
		case JETPACK_MODULE_ACTIVATE_SUCCESS:
			return merge( {}, state, {
				[ siteId ]: {
					[ module ]: false
				}
			} );
		// return initial state when serializing/deserializing
		case SERIALIZE:
		case DESERIALIZE:
			return initialRequestsState;
		default:
			return state;
	}
};
```